src/main/java/picard/util/LiftoverUtils.java

```diff
@@ -0,0 +1,255 @@
+     * This will take an input VariantContext and lift to the provided interval.
+     * If this interval is in the opposite orientation, all alleles and genotypes will be reverse complemented
+     * and indels will be left-aligned.  Currently this is only able to invert biallelic indels, and null will be
+     * returned for any unsupported VC.
+     * @param source The VariantContext to lift
+     * @param target The target interval
+     * @param refSeq The reference sequence, which should match the target interval
+     * @param writeOriginalPosition If true, INFO field annotations will be added to store the original position and contig
+     * @return The lifted VariantContext.  This will be null if the input VariantContext could not be lifted.
+     */
+    public static VariantContext liftVariant(VariantContext source, Interval target, ReferenceSequence refSeq, boolean writeOriginalPosition){
+        if (target == null) {
+            return null;
+        }
+
+        final VariantContextBuilder builder;
+        if (target.isNegativeStrand()){
+            builder = reverseComplementVariantContext(source, target, refSeq);
+        }
+        else {
+            builder = liftSimpleVariantContext(source, target);
+        }
+
+        if (builder == null){
+            return null;
+        }
+
+        builder.filters(source.getFilters());
+        builder.log10PError(source.getLog10PError());
+        builder.attributes(source.getAttributes());
+        builder.id(source.getID());
+
+        if (writeOriginalPosition) {
+            builder.attribute(LiftoverVcf.ORIGINAL_CONTIG, source.getContig());
+            builder.attribute(LiftoverVcf.ORIGINAL_START, source.getStart());
+        }
+
+        return builder.make();
+    }
+
+    protected static VariantContextBuilder liftSimpleVariantContext(VariantContext source, Interval target){
+        if (target == null || source.getReference().length() != target.length()) {
+            return null;
+        }
+
+        // Build the new variant context
+        final VariantContextBuilder builder = new VariantContextBuilder(source);
+        builder.chr(target.getContig());
+        builder.start(target.getStart());
+        builder.stop(target.getEnd());
+
+        return builder;
+    }
+
+    protected static VariantContextBuilder reverseComplementVariantContext(VariantContext source, Interval target, ReferenceSequence refSeq){
+        if (target.isPositiveStrand()){
+            throw new IllegalArgumentException("This should only be called for negative strand liftovers");
+        }
+
+        //not currently supported
+        if (source.isIndel() && !source.isBiallelic()){
+            return null;
+        }
+
+        List<Allele> origAlleles = new ArrayList<>(source.getAlleles());
+        VariantContextBuilder vcb = new VariantContextBuilder(source);
+        vcb.chr(target.getContig());
+
+        final boolean addToStart = source.isIndel() && target.getStart() > 1;
```

Comment:
now that this boolean is going to be used over 3 functions, we might want to document what it is better....

Please note that when an indel is reverse-complemented the allele needs to be shifted by one base to the left. However, this cannot be done if the indel ends up being at the beginning of the contig after the RC. So, this boolean's job is to protect against such a, admitably rare, event.

Reply:
ok